lua/nvim_lsp/configs.lua

```diff
@@ -33,26 +33,26 @@ function configs.__newindex(t, config_name, config_def)
   -- The config here is the one which will be instantiated for the new server,
   -- which is why this is a function, so that it can refer to the settings
   -- object on the server.
-  local function add_callbacks(config)
-    config.callbacks["window/logMessage"] = function(err, method, params, client_id)
+  local function add_handlers(config)
+    config.handlers["window/logMessage"] = function(err, method, params, client_id)
       if params and params.type <= config.log_level then
         -- TODO(ashkan) remove this after things have settled.
-        assert(lsp.callbacks, "Update to Nvim HEAD. This is an incompatible interface.")
-        assert(lsp.callbacks["window/logMessage"], "Callback for window/logMessage notification is not defined")
-        lsp.callbacks["window/logMessage"](err, method, params, client_id)
+        assert(not lsp.callbacks, "lsp.callbacks has been deprecated. See here for more: https://github.com/nvim-lua/diagnostic-nvim/issues/73")
```

Comment:
I think this should be moved out of the inner function. This should be on line 36 I think (where we add the handlers).

We don't need to check this every time.

Reply:
Also, it should reference this PR: https://github.com/neovim/neovim/pull/12655 Which is the one with the most information and most context.

Reply:
I moved the line out, although now every time I initiate an LSP it causes an error. The repo doesn't have any more mentions of callbacks, so is this working right?

Reply:
Maybe what we should do is check this each time a new config is registered (to make sure no deprecated settings are being added).

I'll look into that more tonight, since it looks like `callbacks` is getting populated still somehow.